src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseEventEnablingService.java

```diff
@@ -24,4 +33,24 @@ public Boolean isEventEnabled(String enablingCondition,
         }
         return Boolean.TRUE;
     }
+
+    public Boolean isEventEnabled(String enablingCondition,
+                                  List<CaseViewField> caseViewFields) {
+        if (StringUtils.isNotEmpty(enablingCondition)) {
+            Map<String, Object> caseViewFieldData = caseViewFields.stream()
+                .filter(injectedDataCaseViewField -> injectedDataCaseViewField.getId().startsWith("[INJECTED_DATA"))
+                .collect(
+                    Collectors.toMap(
+                        caseViewField -> replaceDotDelimiter(caseViewField.getId()),
+                        caseViewField -> TextNode.valueOf((String) caseViewField.getValue())));
```

Comment:
```suggestion
                        CaseViewField::getValue));
```
Wondering whether we need to bother with casting and creating a `TextNode`? It looks like later logic within `enablingConditionParser.evaluate()` already converts objects to appropriate `JsonNode` implementations.

Reply:
Fixed in https://github.com/hmcts/ccd-data-store-api/pull/1985/commits/e9775ac2efe226ec855e75b4e136f32f16967b80